packages/core/src/priority-queue/index.ts

```diff
@@ -7,22 +6,20 @@ import { backoff } from './backoff'
  */
 export const ON_REMOVE_FROM_FUTURE = 'onRemoveFromFuture'
 
-export interface WithID {
+interface QueueItem {
   id: string
 }
 
-export class PriorityQueue<
-  Context extends WithID = CoreContext
-> extends Emitter {
-  protected future: Context[] = []
-  protected queue: Context[]
+export class PriorityQueue<Item extends QueueItem = QueueItem> extends Emitter {
```

Comment:
There are no logic changes, just typescript changes to clarify that PriorityQueue is completely decoupled from the analytics domain, as this has proven to be confusing to a few people (me included).